### ISSUE TITLE
C1 does not power back on after standby mode

### ISSUE THREAD

**Issue by Amoeba00** [Maassoft/ColorControl#61]
48 C1 with 3.20.80 firmware, Win10 21H2, nVidia 2070S, latest drivers (no GeForce experience installed), CC v5.0.1.0
Screen Saver: Blank after 2 minutes
Power Management: After 3 minutes, power off display
All of the LG controller options are enabled (except the power off on shutdown).
Under Options-General, everything checked and LG Controller set to default of 10 retries.
Testing the power off/on works, Use old WOL is not checked.

Issue:  TV does not power back on after standby mode.  Need to use the remote to power back on.

Am I missing a setting or something else?  

Thanks!

Edit:  Updated utility to most recent version and then discovered that the utility isn't reading the new IP from the TV - and probably why this isn't working properly.  Opened a separate ticket for that item and will update here if that fixes it.

**Comment by Amoeba00** [comment]
Getting new IP required manual editing of registry, but the TV still doesn't turn back on after resuming from standby.  Test works properly - so I don't know what else to try unless someone else has suggestions on what Win10 settings to set.  Thanks!

**Comment by Maassoft** [comment]
Sorry, somehow I missed this issue, but I think it might have something to do with the system showing a lock screen after resume. Do you have to enter a password after the system resumes from standby?

**Comment by Amoeba00** [comment]
No worries.  Yes, I do have to enter a password after system resumes.

**Comment by Maassoft** [comment]
Okay, I've tested this and it does work on my system. Maybe you log off first before you let your system sleep?
My app is a user app and only works when a user is logged in, LGTVCompanion runs as a Windows service and can function even when no user is logged on, so that could be why it works for that one.

**Comment by Amoeba00** [comment]
So oddly enough - I switched to Windows 11 (new drive, clean install) - and now LGTV Companion doesn't resume after sleep either.  I don't log off my profile, but the password prompt is there to log back on.  Does it work if you just "lock" your profile and not log off?

However, since nothing it working at the moment - I'll try ColorControl again.  I've set the system timers to sleep after 5 min - so will report results here shortly.

**Comment by Maassoft** [comment]
That's odd indeed. But are you saying you have your system configured to automatically sleep after a timeout (while you've locked your profile)? I only tested with manually sleep so that could be different.
Is there anything in the log after resume that the app is at least trying to wake the tv?
Here's an example of my log after resume:
````
2022-01-27 18:52:35.9456|DEBUG|ColorControl.LgService|PowerModeChanged: Resume
2022-01-27 18:52:36.0113|DEBUG|ColorControl.LgDevice|WakeAndConnectWithRetries: attempt 1 of 10...
2022-01-27 18:52:36.0417|DEBUG|ColorControl.WOL|Found network: Ethernet 17 (Intel(R) I211 Gigabit Network Connection), type: Ethernet
2022-01-27 18:52:36.0417|DEBUG|ColorControl.WOL|Found network: Ethernet (Fortinet Virtual Ethernet Adapter (NDIS 6.30)), type: Ethernet
2022-01-27 18:52:36.0417|DEBUG|ColorControl.LgDevice|WOL failed
2022-01-27 18:52:36.5752|DEBUG|ColorControl.LgService|Process monitor: test connection with [LG] webOS TV OLED55C9PLA...
2022-01-27 18:52:37.3598|DEBUG|ColorControl.LgDevice|Cannot apply LG-preset: no connection could be made
2022-01-27 18:52:37.3598|DEBUG|ColorControl.LgService|Process monitor: test connection result: False
2022-01-27 18:52:38.0167|DEBUG|ColorControl.LgDevice|WakeAndConnectWithRetries: attempt 2 of 10...
2022-01-27 18:52:38.0376|DEBUG|ColorControl.WOL|Found network: Ethernet 17 (Intel(R) I211 Gigabit Network Connection), type: Ethernet
2022-01-27 18:52:38.0376|DEBUG|ColorControl.WOL|Found network: Ethernet (Fortinet Virtual Ethernet Adapter (NDIS 6.30)), type: Ethernet
2022-01-27 18:52:38.0376|DEBUG|ColorControl.LgDevice|WOL failed
2022-01-27 18:52:38.3639|DEBUG|ColorControl.LgService|Process monitor: test connection with [LG] webOS TV OLED55C9PLA...
2022-01-27 18:52:38.8802|DEBUG|ColorControl.LgDevice|Cannot apply LG-preset: no connection could be made
2022-01-27 18:52:38.8802|DEBUG|ColorControl.LgService|Process monitor: test connection result: False
2022-01-27 18:52:39.8861|DEBUG|ColorControl.LgService|Process monitor: test connection with [LG] webOS TV OLED55C9PLA...
2022-01-27 18:52:40.0255|DEBUG|ColorControl.LgDevice|WakeAndConnectWithRetries: attempt 3 of 10...
2022-01-27 18:52:40.0448|DEBUG|ColorControl.WOL|Found network: Ethernet 17 (Intel(R) I211 Gigabit Network Connection), type: Ethernet
2022-01-27 18:52:40.0448|DEBUG|ColorControl.WOL|Found network: Ethernet (Fortinet Virtual Ethernet Adapter (NDIS 6.30)), type: Ethernet
2022-01-27 18:52:40.0448|DEBUG|ColorControl.LgDevice|WOL failed
2022-01-27 18:52:40.3864|DEBUG|ColorControl.LgDevice|Cannot apply LG-preset: no connection could be made
2022-01-27 18:52:40.3864|DEBUG|ColorControl.LgService|Process monitor: test connection result: False
2022-01-27 18:52:41.3919|DEBUG|ColorControl.LgService|Process monitor: test connection with [LG] webOS TV OLED55C9PLA...
2022-01-27 18:52:41.9076|DEBUG|ColorControl.LgDevice|Cannot apply LG-preset: no connection could be made
2022-01-27 18:52:41.9076|DEBUG|ColorControl.LgService|Process monitor: test connection result: False
2022-01-27 18:52:42.0331|DEBUG|ColorControl.LgDevice|WakeAndConnectWithRetries: attempt 4 of 10...
2022-01-27 18:52:42.0549|DEBUG|ColorControl.WOL|Found network: Ethernet 17 (Intel(R) I211 Gigabit Network Connection), type: Ethernet
2022-01-27 18:52:42.0549|DEBUG|ColorControl.WOL|Broadcast WOL in network: Ethernet 17 (Intel(R) I211 Gigabit Network Connection), local address: 192.168.178.25
2022-01-27 18:52:42.0549|DEBUG|ColorControl.WOL|Found network: Ethernet (Fortinet Virtual Ethernet Adapter (NDIS 6.30)), type: Ethernet
2022-01-27 18:52:42.0549|DEBUG|ColorControl.LgDevice|WOL succeeded
2022-01-27 18:52:42.9130|DEBUG|ColorControl.LgService|Process monitor: test connection with [LG] webOS TV OLED55C9PLA...
2022-01-27 18:52:52.9809|DEBUG|ColorControl.LgDevice|[[LG] webOS TV OLED55C9PLA] Power state change: {"subscribed":true,"returnValue":true,"state":"Active"}
2022-01-27 18:52:52.9809|DEBUG|ColorControl.LgDevice|PoweredOffBy: Unknown, PoweredOffViaApp: False
2022-01-27 18:52:53.0129|DEBUG|ColorControl.LgDevice|PictureSettingsChanged: {"contrast":"95","backlight":"10","brightness":"51","color":"47"}
2022-01-27 18:52:53.0129|DEBUG|ColorControl.LgDevice|Connect succeeded: True
````

**Comment by Amoeba00** [comment]
So one thing to report with LGTV Companion - is that I had to change the type of WOL magic packet.  Originally, I had it set to the  "Device IP-address" - which didn't work if the TV was in "off" mode.  That dev suggested I change it to the "Broadcast address according to the device IP-address and subnet mask" - which just means add the network subnet mask to the WOL packet.  Now it works.

I'll go back now to CC, check the log a bit closer, and report back.

Win 11 Power Settings: Screen set to off at 3min and Sleep mode begins at 5min

Out of the 5 power settings - I have enabled 2, 4, and 5.
![image](https://user-images.githubusercontent.com/39176113/151425655-a08b5ed9-2934-42bd-8498-ea2b126e7c24.png)

**Comment by Amoeba00** [comment]
Well, oddly enough now CC works too.  It took 30 seconds this time - 4 WOL packets according to the log (same as yours) before the TV clicked on, but good to know that it works.  I'll see if it works after a night of sleep and report back.

**Comment by Amoeba00** [comment]
So it doesn't appear to resume if I manually power off the TV - here's my log on that.  
```
2022-01-28 08:13:53.8457|DEBUG|ColorControl.LgService|PowerModeChanged: Resume
2022-01-28 08:13:53.8457|DEBUG|ColorControl.LgService|[[LG] webOS TV OLED48C1PUB]: device was manually powered off by user, not powering on
2022-01-28 08:14:36.6845|ERROR|LgTv.LgTvApiCore|Connection_MessageReceived: status: Timeout, exception: One or more errors occurred. (0x80072EE2)
Inner exception: 0x80072EE2
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at LgTv.LgTvApiCore.Connection_MessageReceived(MessageWebSocket sender, MessageWebSocketMessageReceivedEventArgs args) in C:\Users\vinni\source\repos\ColorControl\ColorControl\lgtv\LgTvConnection.cs:line 240
2022-01-28 08:14:36.6951|DEBUG|ColorControl.LgService|Process monitor: TV(s) where connected, but not any longer
2022-01-28 08:14:37.7004|DEBUG|ColorControl.LgService|Process monitor: test connection with [LG] webOS TV OLED48C1PUB...
2022-01-28 08:14:43.2210|DEBUG|ColorControl.LgDevice|Cannot apply LG-preset: no connection could be made
2022-01-28 08:14:43.2210|DEBUG|ColorControl.LgService|Process monitor: test connection result: False
2022-01-28 08:14:44.2392|DEBUG|ColorControl.LgService|Process monitor: test connection with [LG] webOS TV OLED48C1PUB...
2022-01-28 08:14:49.7659|DEBUG|ColorControl.LgDevice|Cannot apply LG-preset: no connection could be made
2022-01-28 08:14:49.7659|DEBUG|ColorControl.LgService|Process monitor: test connection result: False
2022-01-28 08:14:54.3497|DEBUG|ColorControl.LgService|Process monitor: test connection with [LG] webOS TV OLED48C1PUB...
2022-01-28 08:14:59.8700|DEBUG|ColorControl.LgDevice|Cannot apply LG-preset: no connection could be made
2022-01-28 08:14:59.8700|DEBUG|ColorControl.LgService|Process monitor: test connection result: False
2022-01-28 08:15:00.8797|DEBUG|ColorControl.LgService|Process monitor: test connection with [LG] webOS TV OLED48C1PUB...
2022-01-28 08:15:06.4101|DEBUG|ColorControl.LgDevice|Cannot apply LG-preset: no connection could be made
2022-01-28 08:15:06.4101|DEBUG|ColorControl.LgService|Process monitor: test connection result: False
2022-01-28 08:15:07.4156|DEBUG|ColorControl.LgService|Process monitor: test connection with [LG] webOS TV OLED48C1PUB...
2022-01-28 08:15:12.9230|DEBUG|ColorControl.LgDevice|Cannot apply LG-preset: no connection could be made
2022-01-28 08:15:12.9230|DEBUG|ColorControl.LgService|Process monitor: test connection result: False
2022-01-28 08:15:13.9381|DEBUG|ColorControl.LgService|Process monitor: test connection with [LG] webOS TV OLED48C1PUB...
2022-01-28 08:15:14.0271|DEBUG|ColorControl.LgDevice|[[LG] webOS TV OLED48C1PUB] Power state change: {"subscribed":true,"returnValue":true,"state":"Active"}
2022-01-28 08:15:14.0271|DEBUG|ColorControl.LgDevice|PoweredOffBy: Unknown, PoweredOffViaApp: False
2022-01-28 08:15:14.0412|DEBUG|ColorControl.LgDevice|PictureSettingsChanged: {"contrast":75,"backlight":75,"brightness":50,"color":"55"}
2022-01-28 08:15:14.0492|DEBUG|ColorControl.LgService|Process monitor: test connection result: True
2022-01-28 08:15:14.0492|DEBUG|ColorControl.LgService|Process monitor: TV(s) where not connected, but connection has now been established
2022-01-28 08:15:24.6292|DEBUG|ColorControl.ADLWrapper|Number Of Adapters: 8
```

**Comment by Maassoft** [comment]
Yes, I should've thought of that: if you manually turn off the tv it won't turn on automatically on resume/startup/end of screensaver. That's more of a feature than a bug. But I think in some situations you might want it to turn on anyway, so I think I'll add a toggle to allow powering on the tv even after a manual power off.

**Comment by Maassoft** [comment]
I've just released 5.4.0.0 which has an option to turn on the tv even after manual power off.

**Comment by Amoeba00** [comment]
Yup.  This appears to work after manual power off.  Thanks again!
